pkg/codegen/dotnet/templates.go

```diff
@@ -90,11 +85,11 @@ namespace {{.Namespace}}
         private readonly static string version;
         public static string Version => version;
 
-        static Utilities()
+        static {{.ClassName}}()
         {
-            var assembly = typeof(Utilities).GetTypeInfo().Assembly;
+            var assembly = global::System.Reflection.IntrospectionExtensions.GetTypeInfo(typeof({{.ClassName}})).Assembly;
```

Comment:
`System.Type` already has an `Assembly` property. I'm not sure what this indirection through `GetTypeInfo` is doing?

Reply:
Not sure, but I didn't want to change the semantics.

Reply:
Fair enough, one to re-visit but I'm like 99.9% certain it won't change the semantics here.